tests/Handler/CurlFactoryTest.php

```diff
@@ -254,6 +254,28 @@ public function testAddsCertWithPassword()
         self::assertEquals('test', $_SERVER['_curl'][\CURLOPT_SSLCERTPASSWD]);
     }
 
+    public function testAddsDerCert()
+    {
+        $certFile = tempnam(sys_get_temp_dir(), "mock_test_cert");
+        rename($certFile, $certFile .= '.der');
+        $f = new Handler\CurlFactory(3);
+        $f->create(new Psr7\Request('GET', Server::$url), ['cert' => $certFile]);
+        self::assertArrayHasKey(\CURLOPT_SSLCERTTYPE, $_SERVER['_curl']);
+        self::assertEquals('DER', $_SERVER['_curl'][\CURLOPT_SSLCERTTYPE]);
+        \unlink($certFile);
```

Comment:
We dont want to fail the test because we cannot clean up after ourselves. 


```suggestion
        $certFile = tempnam(sys_get_temp_dir(), "mock_test_cert");
        rename($certFile, $certFile .= '.der');
        try {
            $f = new Handler\CurlFactory(3);
            $f->create(new Psr7\Request('GET', Server::$url), ['cert' => $certFile]);
            self::assertArrayHasKey(\CURLOPT_SSLCERTTYPE, $_SERVER['_curl']);
            self::assertEquals('DER', $_SERVER['_curl'][\CURLOPT_SSLCERTTYPE]);
        } finally {
            \@unlink($certFile);
        }
```

Reply:
true, but I just followed the model from `StreamHandlerTest`

Reply:
... but I've pushed the suggested change nevertheless, let's get this merged :)